packages/lib-classifier/src/components/Classifier/components/MetaTools/MetaTools.js

```diff
@@ -31,69 +21,54 @@ function storeMapper(store) {
   }
 }
 
-class MetaTools extends Component {
-  constructor () {
-    super()
-
-    this.addToCollection = this.addToCollection.bind(this)
-  }
+const MetaTools = () => {
+  const {
+    className = '',
+    interactionTask = {},
+    isThereMetadata = false,
+    screenSize,
+    subject = null,
+    upp = null
+  } = useStores(storeMapper)
+  const { shownMarks, marks, togglePreviousMarks, type } = interactionTask
+  const gap = (screenSize === 'small') ? 'xsmall' : 'small'
+  const margin = (screenSize === 'small') ? { top: 'small' } : 'none'
 
-  addToCollection () {
-    const { subject } = this.props
+  const addToCollection = () => {
     subject.addToCollection()
   }
 
-  // TODO: Add fallbacks for when Panoptes is not serializing the subject favorite info
-  render () {
-    const {
-      className,
-      interactionTask,
-      isThereMetadata,
-      screenSize,
-      subject,
-      upp
-    } = this.props
-    const { shownMarks, marks, togglePreviousMarks, type } = interactionTask
-    const gap = (screenSize === 'small') ? 'xsmall' : 'small'
-    const margin = (screenSize === 'small') ? { top: 'small' } : 'none'
-
-    return (
-      <Box
-        key={subject && subject.id}
-        className={className}
-        direction='row-responsive'
-        gap={gap}
-        margin={margin}
-      >
-        <Metadata isThereMetadata={isThereMetadata} metadata={subject && subject.metadata} />
-        <FavouritesButton
-          checked={subject?.favorite}
-          disabled={!subject || !upp}
-          onClick={subject?.toggleFavorite}
+  return (
+    <Box
+      key={subject && subject.id}
+      className={className}
+      direction='row-responsive'
+      gap={gap}
+      margin={margin}
+    >
+      <Metadata
+        isThereMetadata={isThereMetadata}
+        metadata={subject && subject.metadata}
+      />
+      <FavouritesButton
+        checked={subject?.favorite}
+        disabled={!subject || !upp}
+        onClick={subject?.toggleFavorite}
+      />
+      <CollectionsButton
+        disabled={!subject || !upp}
+        onClick={addToCollection}
+      />
```

Comment:
When I run locally, signed in, and load https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats/classify/workflow/693 or https://local.zooniverse.org:3000/projects/pmlogan/poets-and-lovers/classify/workflow/21362/subject-set/104809?env=production the Metadata, FavouritesButton, and CollectionsButton are unexpectedly disabled. Console logging the subject from MetaTools returns `null` and the component doesn't appear to rerender once the subject (and related isThereMetadata) is defined.

Reply:
You're right! I think adding `observer` restored some of the expected functionality, but I also noticed that you can visit [i-fancy-cats on frontend.preview](https://frontend.preview.zooniverse.org/projects/brooke/i-fancy-cats?env=staging), sign-in to enable FavouritesButton and CollectionsButton, but then upon sign-out the `upp` from the store still contains user data, and the buttons are all still enabled. I think this is a bug of the UserProjectPreferences store so I'll do a quick search for an already existing Github Issue or open a new one.